test/membrane/rtp/session_bin_test.exs

```diff
@@ -28,6 +32,46 @@ defmodule Membrane.RTP.Session.ReceiveBinTest do
 
   @fmt_mapping %{96 => {:H264, 90_000}, 127 => {:MPA, 90_000}}
 
+  @spec inspect_packet_fields(Packet.t(), %{}) :: boolean()
+  defp inspect_packet_fields(packet, fields_values), do: compare(packet, fields_values)
+
+  @spec compare(any(), any()) :: boolean()
+  defp compare(left, right) when is_map(left) or is_struct(left) do
+    if is_map(right) do
+      right
+      |> Enum.all?(fn {k, v} ->
+        case Map.has_key?(left, k) do
+          true ->
+            compare(Map.get(left, k), v)
+
+          _ ->
+            false
+        end
+      end)
+    else
+      false
+    end
+  end
+
+  defp compare(left, right), do: left == right
```

Comment:
```suggestion
  @spec compare_common_fields(any(), any()) :: boolean()
  defp compare_common_fields(left, right) when is_map(left) do
    is_map(right) and Enum.all?(right, fn {k, v} ->
        Map.has_key?(left, k) and compare_common_fields(Map.get(left, k), v)
    end)
  end

  defp compare_common_fields(left, right), do: left == right
```

Reply:
Nice, that's much more compact :D But here in line 39 `defp compare_common_fields(left, right) when is_map(left) do` there should be also `or is_struct(left)` as the left side is any structure(map or struct) and right is map of fields we want to match. Do you agree or getting rid of `or is_struct(left)` was intended?